corehq/apps/auditcare/middleware.py

```diff
@@ -58,16 +58,20 @@ def __call__(self, request):
         We also need to add the user field when it was not initially inferred from the sessionid,
         such as when using Api Key, Basic Auth, Digest Auth, or HMAC auth.
         """
-        response = self.get_response(request)
-        audit_doc = getattr(request, 'audit_doc', None)
-        if audit_doc:
-            if not audit_doc.user:
-                if hasattr(request, 'audit_user'):
-                    audit_doc.user = request.audit_user
-                elif hasattr(request, 'couch_user'):
-                    audit_doc.user = request.couch_user.username
-            audit_doc.status_code = response.status_code
-            audit_doc.save()
+        response = None
+        try:
+            response = self.get_response(request)
+        finally:
+            audit_doc = getattr(request, 'audit_doc', None)
+            if audit_doc:
+                if not audit_doc.user:
+                    if hasattr(request, 'audit_user'):
+                        audit_doc.user = request.audit_user
+                    elif hasattr(request, 'couch_user'):
+                        audit_doc.user = request.couch_user.username
+                if response is not None:
+                    audit_doc.status_code = response.status_code
+                audit_doc.save()
```

Comment:
Nit: I get sketched out whenever I see long/complicated logic within exception handling. This looks logically sound, but I think I would be less concerned if the `finally` clause was extracted out to a helper method:
```
try:
  response = self.get_response(request)
finally:
  self._update_audit(request, response)